include/ttmlir/Target/Utils/MLIRToFlatbuffer.h

```diff
@@ -128,19 +128,24 @@ toFlatbuffer(FlatbufferObjectCache &, ChipCapabilityAttr capabilityAttr) {
 }
 
 inline ::tt::target::ChipCoord toFlatbuffer(FlatbufferObjectCache &cache,
-                                            ChipCoordAttr chipCoord) {
+                                            const ChipCoordAttr &chipCoord) {
   return ::tt::target::ChipCoord(chipCoord.getRack(), chipCoord.getShelf(),
                                  chipCoord.getY(), chipCoord.getX());
 }
 
-inline ::tt::target::ChipChannel toFlatbuffer(FlatbufferObjectCache &cache,
-                                              ChipChannelAttr chipChannel) {
-  return ::tt::target::ChipChannel(chipChannel.getEndpoint0(),
-                                   chipChannel.getEndpoint1());
+inline ::tt::target::ChipChannel
+toFlatbuffer(FlatbufferObjectCache &cache, const ChipChannelAttr &chipChannel) {
+  return ::tt::target::ChipChannel(
+      chipChannel.getDeviceId0(),
+      ::tt::target::Dim2d(chipChannel.getEthernetCoreCoord0()[0],
+                          chipChannel.getEthernetCoreCoord0()[1]),
+      chipChannel.getDeviceId1(),
+      ::tt::target::Dim2d(chipChannel.getEthernetCoreCoord1()[0],
+                          chipChannel.getEthernetCoreCoord1()[1]));
 }
 
 inline ::tt::target::Dim2d toFlatbuffer(FlatbufferObjectCache &cache,
-                                        GridAttr arch) {
+                                        const GridAttr &arch) {
```

Comment:
Not blocking the review, but typically we only pass the MLIR types, i.e. `Value`, `*Attr`, `*Type`, by value.  MLIR already enforces that they are wrappers around some underlying storage and it is convention to treat them as value types.  It doesn't hurt to pass it by `const&`, but it's not necessary.  All MLIR types are also immutable, so they are truly value types in this sense.

Reply:
Ahhh I see, thanks for the explanation, I'll undo these changes!